src/grid/api.service.ts

```diff
@@ -51,12 +51,13 @@ export class IgxGridAPIService {
         }
     }
 
-    public get_row(id: string, index: number): IgxGridRowComponent {
-        return this.get(id).rowList.find((row) => row.index === index);
+    public get_row(id: string, rowSelector: any): IgxGridRowComponent {
+        const primaryKey = this.get(id).primaryKey;
+        return this.get(id).rowList.find((row) => primaryKey ? row.rowData[primaryKey] === rowSelector : row.index === rowSelector);
```

Comment:
I wouldn't do it this way - yes it looks cleaner as a one-liner but you are checking if primaryKey is set for each record instead of checking it once outside and running different finds based on the result.
@rkaraivanov - not sure if we support two dimensional arrays for igxGrid, but if we do and the primaryKey is the 0-index column, then the check will fail as well ( if (0) = false )

Reply:
Reworked in the latest commit